test/pylib/util.py

```diff
@@ -131,6 +130,13 @@ async def find_host():
     return await wait_for(find_host, deadline)
 
 
+async def safe_gather(*awaitables: Awaitable, return_exceptions: bool = True):
+    results = await asyncio.gather(*awaitables, return_exceptions=return_exceptions)
```

Comment:
the whole idea of safe_gather run gather always with return_exceptions=True. otherwise we are losing the main idea of safety. 

```
async def safe_gather(*awaitables: Awaitable):
    results = await asyncio.gather(*awaitables, return_exceptions=True)
    for result in results:
        if isinstance(result, BaseException):
            raise result from None
```